Various docs additions (#681)
garden-co/jazz: docs/content/docs/auth/lifecycle.mdx

  in OPFS; React Native uses native storage; memory drivers keep it only for the life of the process.

Clearing one does not automatically clear the other. That separation is intentional: a development
storage reset should not silently destroy a user's identity, and signing out of an auth provider
should not necessarily delete offline data.

## Local-first identity storage

Local-first auth derives the user's Jazz identity from a 32-byte secret. The same secret always
produces the same Jazz user ID, so preserving the secret preserves identity across app restarts.

```ts
const secret = await BrowserAuthSecretStore.getOrCreateSecret();

const db = await createDb({
  appId: "my-app",
  secret,
});
```

`useLocalFirstAuth()` wraps this storage for React and Expo apps. Its `signOut()` method clears the
stored secret, which means the next login without a restored secret creates a different Jazz
identity.

<Callout type="warn">
  The local-first secret is the user's identity. If it is lost, rows owned only by that identity may
  become inaccessible unless the user restores the same secret from a recovery passphrase, passkey

>   backup, or linked external account.

Reviewer: ```suggestion
  The local-first secret is the user's identity. If it is lost, rows owned only by that identity may
  become inaccessible unless the user restores the same secret from a recovery passphrase, passkey
  backup, or linked external account. The secret should be protected carefully: anyone with the secret can
  authenticate as the user. 
```

</Callout>

Use [Local-first auth](/docs/auth/local-first-auth#backing-up-and-restoring-the-secret) to add
recovery before shipping flows that clear or replace a local-first secret.

## Switching users

> Do not use `db.updateAuthToken(null)` to sign out or switch principals on a live `Db`. Recreate the

Reviewer: ```suggestion
Recreate the
```

client with a new auth config whenever the active principal changes:

```tsx
<JazzProvider key={sessionKey} config={config}>
  <App />
</JazzProvider>
```

> For the same external principal, `db.updateAuthToken(freshJwt)` is fine for refreshing an expiring

Reviewer: ```suggestion
For the same external user, `db.updateAuthToken(freshJwt)` is fine for refreshing an expiring
```

> bearer token. For cookie-based auth, `db.updateCookieSession(nextSession)` updates the mirrored

Reviewer: ```suggestion
bearer token. Do not use `db.updateAuthToken(null)` to sign out or switch users on a live `Db`. For cookie-based auth, `db.updateCookieSession(nextSession)` updates the mirrored
```

> session for the same principal while the HttpOnly cookie remains the transport credential.

Reviewer: ```suggestion
session for the same user while the HttpOnly cookie remains the transport credential.
```

## Logout

> Use `db.logout()` when you are done with a `Db` instance during logout or a principal switch. It

Reviewer: ```suggestion
Use `db.logout()` when you are done with a `Db` instance during logout or switching users. It
```
